Autopilot/TelemetryManager/Src/telemetryStateClasses.cpp

```diff
@@ -19,7 +29,9 @@ telemetryState& initialMode::getInstance()
 void obtainDataMode::execute(telemetryManager* telemetryMgr)
 {
     //obtain data from ground
+    XBEE::Receive_GS_Data(); //Receives data in MavLink form from the XBEE
```

Comment:
With singletons, this will not be needed anymore. Instead you will have the XBEE class insatnce as a parameter of the `obtainDataMode` class. You can initialize the parameter in the constructor of the `obtainDataMode` class in the `telemetryStateClasses.hpp` file by calling the getInstance() in XBEE method when you make it

 within the constructor of the `obtainDataMode` class you will call an ins